src/juju_doctor/probes.py

```diff
@@ -0,0 +1,53 @@
+"""Helper module to wrap and execute probes."""
+
+import importlib.util
+import inspect
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+
+from rich.logging import RichHandler
+
+from juju_doctor.artifacts import Artifacts
+
+SUPPORTED_PROBE_TYPES = ["status", "bundle", "show_unit"]
+
+logging.basicConfig(level=logging.WARN, handlers=[RichHandler()])
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class Probe:
+    """A probe that can be executed via juju-doctor."""
+
+    name: str
+    uri: str
+    original_path: Path  # path in the source folder
+    path: Path  # path in the temporary folder
```

Comment:
What's the reasoning for having these as classvars?

Reply:
They were already there:
- `name` is used to print the probe name in the results
- `path` is used to import the probe file
- `uri` and `original_path` are there just in case we want to print them in the future, we can remove them if you want